src/mlpack/methods/neighbor_search/knn_main.cpp

```diff
@@ -209,12 +245,6 @@ int main(int argc, char *argv[])
     Log::Info << "Loaded kNN model from '" << inputModelFile << "' (trained on "
         << knn.Dataset().n_rows << "x" << knn.Dataset().n_cols << " dataset)."
         << endl;
-
-    // Adjust singleMode and naive if necessary.
-    knn.SingleMode() = CLI::HasParam("single_mode");
-    knn.Naive() = CLI::HasParam("naive");
-    knn.LeafSize() = size_t(lsInt);
-    knn.Epsilon() = epsilon;
```

Comment:
A possible approach is this:
"When providing an input model file, all the extra options that are provided will be updated in the given model."
With the only limitation that it is not possible to modify a single tree model to a dual tree model, because we don't have a "--dual_mode" command line option.